tests/python/relax/test_transform_cutlass_codegen.py

```diff
@@ -0,0 +1,392 @@
+                (D,) = df.output_vars
+                R.func_ret_value(D)
+    relax_mod = ib.get()
+    return relax_mod
+
+
+def test_cutlass_batch_dense2_bias():
+    b, m, n, k = 2, 128, 128, 128
+    build(constructBatchGEMM2_bias(b, m, n, k))
+    dev = tvm.cuda()
+    A = np.random.rand(b, m, k).astype("float16") * 5
+    B = np.random.rand(b, k, n).astype("float16") * 5
+    bias = np.random.rand(1, n).astype("float16") * 20
+    A_tvm = tvm.nd.array(A, dev)
+    B_tvm = tvm.nd.array(B, dev)
+    bias_tvm = tvm.nd.array(bias, dev)
+    executable = tvm.runtime.load_module(PKG_FILE)
+    result = f_run(executable, dev, A_tvm, B_tvm, bias_tvm)
+    np.testing.assert_allclose(result.numpy(), A @ B + bias, rtol=1e-2)
+
+
+def constructConv2D(N, C, H, W, KH, KW, O, strides, padding, dilation):
+    from tvm.script.ir_builder import IRBuilder
+    from tvm.script.ir_builder import ir as I
+    from tvm.script.ir_builder import relax as R
+    from tvm.script.ir_builder import tir as T
+
+    with IRBuilder() as ib:  # pylint: disable=invalid-name
+        with I.ir_module() as frame:
+            with R.function():
+                R.func_name("main")
+                x = R.arg(
+                    "x", relax.TensorStructInfo((N, H, W, C), A_TYPE)
+                )  # pylint: disable=invalid-name
+                w = R.arg(
+                    "w", relax.TensorStructInfo((O, KH, KW, C), B_TYPE)
+                )  # pylint: disable=invalid-name
+                C = R.nn.conv2d(
+                    x,
+                    w,
+                    kernel_size=(KH, KW),
+                    strides=strides,
+                    padding=padding,
+                    dilation=dilation,
+                    groups=1,
+                    channels=None,
+                    data_layout="NHWC",
+                    kernel_layout="OHWI",
+                    out_layout="NHWC",
+                    out_dtype=C_TYPE,
+                )
+                R.func_ret_value(C)
+    mod = ib.get()
+    return mod
+
+
+def test_cutlass_conv2d():
+    n, c, h, w = 1, 3, 224, 224
+    kh, kw, o = 3, 3, 64
+    # strides = (1, 1)
+    # padding = (3, 3)
+    # dilation = (1, 1)
+    counter = 0
+    for strides in [(1, 1), (2, 2)]:
+        for padding in [(0, 0), (3, 3)]:
+            for dilation in [(1, 1), (4, 4)]:
+                filename = "/tmp/" + "test_transform_cutlass_codegen" + str(counter) + ".so"
+                build(constructConv2D(n, c, h, w, kh, kw, o, strides, padding, dilation), filename)
+                dev = tvm.cuda()
+                np.random.seed(0)
+                A = np.random.rand(n, h, w, c).astype("float16") * 5
+                B = np.random.rand(o, kh, kw, c).astype("float16") * 5
+                A_tvm = tvm.nd.array(A, dev)
+                B_tvm = tvm.nd.array(B, dev)
+                executable = tvm.runtime.load_module(filename)
+                result = f_run(executable, dev, A_tvm, B_tvm)
```

Comment:
use before /expected tests ideally instead of execution test. exection test should go to a separate place under test_integration_trasnform_cutlass_codegen.py

Reply:
I can't think of a way to add before/expected tests because after applying the pass there will be extern funcs, and extern funcs cannot be parsed